File: ConvexCodeExamplesExternal/src/main/JS/amqp/consumeC114.js
#!/usr/bin/env node

var amqp = require('amqplib/callback_api');

amqp.connect('amqps://01OZLElajKpQJonw@amqp.convexglobal.io:5671/stream', function(error0, connection) {
  if (error0) {
    throw error0;
  }
  connection.createChannel(function(error1, channel) {
    if (error1) {
      throw error1;
    }
    var exchange = '2cc2aa';

    channel.assertExchange(exchange, 'topic', {
      durable: true
    });

    channel.assertQueue(exchange, {
      durable: true
    }, function(error2, q) {
      if (error2) {
        throw error2;
      }
      console.log(" [*] Waiting for messages in %s. To exit press CTRL+C", q.queue);
      channel.bindQueue(q.queue, exchange, '');

      channel.consume(q.queue, function(msg) {
        if(msg.content) {
            console.log(" [x] %s", msg.content.toString());
          }
      }, {
        noAck: true
      });
    });
  });
});

File: ConvexCodeExamplesExternal/src/main/Python/amqp/consumeC114.py
#!/usr/bin/env python
import pika
import ssl
import sys
import os

ssl_context = ssl.create_default_context()
ssl_options = pika.SSLOptions(ssl_context, "amqp.convexglobal.io")

def main():

   credentials = pika.PlainCredentials('01OZLElajKpQJonw', '')
   parameters = pika.ConnectionParameters(host='amqp.convexglobal.io', port=5671, virtual_host='stream', credentials=credentials, ssl_options=ssl_options)

   connection = pika.BlockingConnection(parameters)
   channel = connection.channel()

   #channel.queue_bind(queue='2cc2aa', exchange='2cc2aa')

   def callback(ch, method, properties, body):
        print(" [x] Received %r" % body)

   channel.basic_consume(queue='2cc2aa', on_message_callback=callback, auto_ack=True)

   print(' [*] Waiting for messages. To exit press CTRL+C')
   channel.start_consuming()

if __name__ == '__main__':
    try:
        main()
    except KeyboardInterrupt:
        print('Interrupted')
        try:
            sys.exit(0)
        except SystemExit:
            os._exit(0)
